core/src/file/indexer/scan.rs

```diff
@@ -172,8 +172,20 @@ fn prepare_values(
   let metadata = fs::metadata(&file_path)?;
   let location_path = location.path.as_ref().unwrap().as_str();
   // let size = metadata.len();
-  let name = extract_name(file_path.file_stem());
-  let extension = extract_name(file_path.extension());
+  let name;
+  let extension;
+
+  // if the 'file_path' is not a directory, then get the extension and name.
+
+  // if 'file_path' is a directory, set extension to an empty string to avoid periods in folder names
+  // - being interpreted as file extensions
+  if !file_path.is_dir() {
+    extension = extract_name(file_path.extension());
+    name = extract_name(file_path.file_stem());
+  } else {
+    extension = "".to_string();
+    name = extract_name(file_path.file_name());
+  }
```

Comment:
since this is an if-else, should we avoid inverting the `is_dir` call so we don't have to double-negative when reading the code?

```suggestion
  if file_path.is_dir() {
    extension = "".to_string();
    name = extract_name(file_path.file_name());
  } else {
    extension = extract_name(file_path.extension());
    name = extract_name(file_path.file_stem());
  }
```